doc/nrf/protocols/matter/end_product/security.rst

```diff
@@ -1,16 +1,121 @@
+
+An important setting, that depends on the hardware platform in use, is the way of generating the AEAD keys.
+The recommended and the most secure option is to use :ref:`lib_hw_unique_key` (HUK) library.
+HUK support is automatically enabled with the :kconfig:option:`CONFIG_TRUSTED_STORAGE_BACKEND_AEAD_KEY_DERIVE_FROM_HUK` Kconfig option for compatible configurations.
+
+Currently, the :ref:`lib_hw_unique_key` library is not supported for nRF52840 Matter-enabled devices.
+Instead, in this case, AEAD keys are derived with a SHA-256 hash (:kconfig:option:`CONFIG_TRUSTED_STORAGE_BACKEND_AEAD_KEY_HASH_UID`).
+This approach is less secure than using the library for key derivation as it will only provide integrity of sensitive material.
+It is also possible to implement a custom AEAD keys generation method when the :kconfig:option:`CONFIG_TRUSTED_STORAGE_BACKEND_AEAD_KEY_CUSTOM` Kconfig option is selected.
+
+.. note::
+   Using HUK library in the case of nRF52840 is possible, yet it requires employing b0 bootloader that would generate the AEAD key at first boot and store it in the dedicated HUK partition that can be accessed only by the CryptoCell peripheral.
+   Note that adding another partition in the FLASH layout implies breaking the firmware backward compatibility with already deployed devices.
+
+You can find an overview of the Trusted Storage layer configuration supported for each |NCS| Matter-enabled platform in the :ref:`matter_platforms_security_support` section.
+
+.. _matter_platforms_security_support:
+
+Matter platforms security support
+*********************************
+
+The following table summarizes the current security configuration and features supported for Matter-enabled hardware platforms in the |NCS|.
+This is a reference configuration that can be modified in the production firmware by using proper Kconfig settings or implementing custom cryptographic backends.
+
+.. list-table:: Matter platforms security support
+   :widths: auto
+   :header-rows: 1
+
+   * - Platform
+     - Networking backend
+     - Cryptography API
+     - Cryptography backend
+     - ARM TrustZone support
+     - Trusted Storage backend
+     - Trusted Storage AEAD key
+   * - nRF52840 SoC
+     - Thread
+     - PSA
+     - Oberon + CryptoCell [1]_
+     - No
+     - Trusted Storage library
+     - SHA-256 hash
+   * - nRF5340 SoC
+     - Thread
+     - PSA
+     - Oberon + CryptoCell [1]_
+     - Yes
+     - Trusted Storage library
+     - Hardware Unique Key (HUK)
+   * - nRF5340 SoC + nRF7002 companion IC
+     - Wi-Fi
+     - Mbed TLS
+     - Oberon + CryptoCell [1]_
+     - Yes
+     - -
+     - -
+.. [1] The CryptoCell backend is used in parallel with the Oberon to implement only two components:
+         * Random Number Generation (RNG).
+         * AEAD keys derivation driver (only in the case of the Thread networking).
+       For all other cryptographic operations, the Oberon backend is used.
```

Comment:
```suggestion
.. [1] The CryptoCell backend is used in parallel with the Oberon backend.
       It is only used to implement Random Nuber Generation (RNG), and the AED keys derivation driver (only for Thread networking).
       For all other cryptographic operations, the Oberon backend is used.
```
Unfortunately, it seems bullets don't work in the doc build. It would have been clearer with them.